task/thermal/src/control.rs

```diff
@@ -402,20 +406,18 @@ enum ControlResult {
 }
 
 impl ThermalControlState {
-    fn write_temperature(
-        &mut self,
-        index: usize,
-        time_ms: u64,
-        value: Celsius,
-    ) {
+    fn write_temperature(&mut self, index: usize, reading: Reading) {
+        let r = TemperatureReading::Valid {
+            time_ms: reading.timestamp,
+            value: Celsius(reading.value),
```

Comment:
Not related to your changes, but - should `reading.value` be typed instead of a raw `f32`?

Reply:
Not in this case – `Reading` is used by every sensor type, so it's a raw value (and you need to know which sensor populated it to know the units).